File: quartz-energy/src/backend/QuartzEnergy.FracSchedule.Services/FracSchedule/Models/Schedules/Overviews/ScheduleOverviewsRequest.cs
﻿namespace QuartzEnergy.FracSchedule.Services.FracSchedule.Models.Schedules.Overviews
{
    using System.Collections.Generic;

    using QuartzEnergy.Common.DataAnnotations.Attributes.Data;
    using QuartzEnergy.Common.DataAnnotations.Attributes.Numbers;
    using QuartzEnergy.Common.Services.Models.Overviews;
    public sealed class ScheduleOverviewsRequest : OverviewsRequestModel
    {
        public ScheduleOverviewsRequest(
            IEnumerable<int> regions, 
            IEnumerable<int> operators, 
            int? startNextDays,
            string sortBy,
            bool desc,
            int? pageSize,
            int? pageNumber)
            : base(sortBy, desc, pageSize, pageNumber)
        {
            this.Regions = regions;
            this.Operators = operators;
            this.StartNextDays = startNextDays;
        }

        [IntIds]
        public IEnumerable<int> Regions { get; }

        [IntIds]
        public IEnumerable<int> Operators { get; }

        [MinIntValue(0)]
        public int? StartNextDays { get; }
    }
}

File: quartz-energy/src/frontend/src/app/frac-schedule/models/schedules/overviews/schedule-overviews-request.ts
import { OverviewsRequest } from './../../../../common/models/overviews/overviews-request';

export class ScheduleOverviewsRequest extends OverviewsRequest {
    constructor (
        public regions: number[],
        public operators: number[],
        public startNextDays: number,
        public sortBy: string,
        public desc: boolean,
        public pageSize: number,
        public pageNumber: number) {
      super(sortBy, desc, pageSize, pageNumber);
    }
}
